frontend/src/pages/Home/HomePage.js

```diff
@@ -1,7 +1,74 @@
-import React from "react";
-
+import React, { useEffect } from "react";
+import ActivityFeed from "../../components/common/ActivityFeed";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  getStudentActivityLog,
+  getUserDetails,
+} from "../../redux/actions/student";
 const HomePage = () => {
-  return <div></div>;
+  const dispatch = useDispatch();
+  const { activityLogs } = useSelector((state) => state.students);
+  const { userData } = useSelector((state) => state.auth);
+  const profile = useSelector((state) => state.profile);
+
+  useEffect(() => {
+    dispatch(getStudentActivityLog(userData.user_id));
+    dispatch(getUserDetails());
+  }, [dispatch, userData.user_id]);
+
+  const displayLearnings = () => {
+    let total_lessons = 0;
+    let total_answer = 0;
+    activityLogs &&
+      activityLogs.map((log) => {
+        if (log.resourcetype === "StudentLesson") {
+          if (userData.user_id === log.student.id) {
+            total_lessons = log.total_lessons;
+          }
+          total_answer +=
+            userData.user_id === log.student.id && log.total_answer;
+        }
+        return null;
+      });
+
+    return [total_lessons, total_answer];
```

Comment:
For the use case of `map` it is to restructure arrays, not just for iterating.
Maybe what you need is the `filter` function, or just `forEach`? pls try to look unto those